.github/workflows/build-and-test.yml

```diff
@@ -55,6 +55,14 @@ jobs:
     - name: Run tests
       run: ./gradlew test -Dkotest.tags='!Integration'
 
+    - name: Upload test results
+      uses: actions/upload-artifact@v4
+      if: success() || failure()
```

Comment:
Maybe instead use `if: always()`?

Reply:
I intentionally used `success() || failure()` because the step should not run if a user cancels the tests.

Reply:
Interesting... I guess that's worth a code comment? Or, if you don't want to comment twice in the code, a hint in the commit message at least would be great.

Reply:
For reference, [these](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions) are the different status check functions.